src/sentry/api/utils.py

```diff
@@ -48,9 +70,15 @@ def get_date_range_from_params(params, optional=False):
             end = parse_datetime_string(params['end'])
         except InvalidQuery as exc:
             raise InvalidParams(exc.message)
-        if start > end:
-            raise InvalidParams('start must be before end')
     elif optional:
         return None, None
 
+    if start > end:
+        raise InvalidParams('start must be before end')
+
+    delta = end - start
+    if delta < MIN_STATS_PERIOD or delta > MAX_STATS_PERIOD:
+        raise InvalidParams(
+            'Time window must be greater than an hour and less than or equal to 90 days')
```

Comment:
Might make sense to make this string dynamic so that it doesn't break if we change min/max period